tools/nvqpp/nvq++.in

```diff
@@ -43,6 +43,21 @@ function list_targets {
 	exit 0
 }
 
+function query_gpu {
+	if [ -x "$(command -v nvidia-smi)" ]; then
+		# Make sure nvidia-smi works.
+		nvidia-smi -L | grep 'Failed\|Error\|error\|failed' >/dev/null 2>&1
```

Comment:
Does `nvidia-smi` not return a non-zero return code when/if it fails?

Reply:
Here I captured the original logic from the nvidia-*.config files into a function. It explicitly checked for the output of the command (I suspect if it fails to connect to the driver, perhaps such message could occur, again not sure of return code). 
Will check with Alex.